### ISSUE TITLE
ExecutionTests should check for feature support before using new Typed UAV Load formats

### ISSUE THREAD

**Issue by jholewinski** [Microsoft/DirectXShaderCompiler#307]
Some ExecutionTests, for example BasicComputeTest, implicitly require support for new typed UAV load formats, which not all DX12 hardware supports. The test tries to use this feature without checking the hardware caps bits, leading to failures like below:

```
00000121	144.96492004	[7712] Verify: SUCCEEDED(m_support.Initialize())	
00000122	144.96496582	[7712]  	
00000123	144.96528625	[7712] Verify: SUCCEEDED(m_support.CreateInstance(CLSID_DxcCompiler, &pCompiler))	
00000124	144.96531677	[7712]  	
00000125	144.96566772	[7712] Verify: SUCCEEDED(m_support.CreateInstance(CLSID_DxcLibrary, &pLibrary))	
00000126	144.96569824	[7712]  	
00000127	144.96611023	[7712] Verify: SUCCEEDED(pLibrary->CreateBlobWithEncodingFromPinned((LPBYTE)pText, strlen(pText), CP_UTF8, &pTextBlob))	
00000128	144.96612549	[7712]  	
00000129	144.97114563	[7712] Verify: SUCCEEDED(pCompiler->Compile(pTextBlob, L"hlsl.hlsl", pEntryPoint, pTargetProfile, nullptr, 0, nullptr, 0, nullptr, &pResult))	
00000130	144.97119141	[7712]  	
00000131	144.97146606	[7712] Verify: SUCCEEDED(pResult->GetStatus(&resultCode))	
00000132	144.97149658	[7712]  	
00000133	144.97180176	[7712] Verify: SUCCEEDED(resultCode)	
00000134	144.97183228	[7712]  	
00000135	144.97216797	[7712] Verify: SUCCEEDED(pResult->GetResult((IDxcBlob **)ppBlob))	
00000136	144.97221375	[7712]  	
00000137	144.97225952	[7712] D3D12 ERROR: ID3D12Device::CreateComputeShader: Shader uses new Typed UAV Load formats, but the device does not support this. To check for support, check device caps via the CheckFeatureSupport() API [ STATE_CREATION ERROR #622: CREATESHADER_INVALIDBYTECODE]	
00000138	144.97282410	[7712] Error: Verify: SUCCEEDED(pDevice->CreateComputePipelineState(&computePsoDesc, IID_PPV_ARGS(ppComputeState))) - Value (0x80070057) [File: D:\nativeDXIL\DirectXShaderCompiler\tools\clang\unittests\HLSL\ExecutionTest.cpp, Function: ExecutionTest::CreateComputePSO, Line: 578]	
00000139	144.97286987	[7712]  	
00000140	144.98577881	[7712] EndGroup: ExecutionTest::BasicComputeTest [Failed]	
```

**Comment by youngkim93** [comment]
Hi Justin, it turns out that there was an issue with the compiler incorrectly setting the compiler flag for new UAV Load format. This issue should have been fixed with #356. I don't have any hardware to test this personally. Please let me know if you are still getting this issue with the fix. Thanks!

**Comment by jholewinski** [comment]
Looks good on our end ! Thanks!
